src/preface.md

```diff
@@ -2,52 +2,37 @@
-
-## Scope of This Book
-
-In this book, we focus specifically on the `esp-rs` ecosystem. It does **not** serve as:
-
-- A general Rust programming language tutorial: Familiarity with the Rust language is assumed to get you started.
-- An embedded development guide: Prior familiarity with basic embedded development is also assumed.
-- A collection of examples: This book explains the overall process of working with the `esp-rs` ecosystem. More examples can be found [here][examples].
-
-Check the [Additional Resources][resources] section if you are not familiar with any of these concepts.
+This book is intended for readers with some experience in Rust and a basic understanding of embedded development and electronics. You should already be familiar with concepts such as cross-compilation, common digital interfaces (UART, SPI, I2C), memory-mapped peripherals, and interrupts. It is not a general Rust tutorial or an introduction to embedded systems — prior knowledge is expected. Check the [Additional Resources][resources] section if you are not familiar with any of these concepts. For practical examples, see the esp-hal examples see [examples].
 
 [examples]: https://github.com/esp-rs/esp-hal/tree/main/examples
 [resources]: #additional-resources
 
 ## Stability and Availability
 
-The `esp-rs` project is under active development and may be subject to changes as it evolves. While we strive for stability, users should expect periodic modifications as we improve the API, improve performance, and introduce new features. [Modules] that are already stabilized will not be subject to breaking changes, while the `unstable` parts of the drivers are being actively worked on and will likely undergo changes in future releases.
-
-Additionally, the embedded Rust ecosystem is still in its infancy; some dependencies, tools, or language features may change over time. We encourage developers to stay up to date with the latest releases and participate in discussions to help shape the future of Rust on Espressif platforms.
-
-[modules]: https://docs.espressif.com/projects/rust/esp-hal/1.0.0-beta.0/esp32c6/esp_hal/index.html#modules
+The `esp-rs` project is under active development and may be subject to changes as it evolves. While we strive for stability, users should expect periodic modifications as we improve the API, enhance performance, and introduce new features. Modules (see the latest) that are already stabilized will not be subject to breaking changes, in accordance with semantic versioning - `SemVer`. However, `unstable` features—such as parts of the HAL and certain drivers—are actively being developed and are not covered by `SemVer` guarantees. This means that using these `unstable` components may break your project with a simple `cargo update`, much like working with Rust's `nightly` compiler. This kind of instability is common across the broader Rust embedded ecosystem, which is still rapidly evolving. Expect frequent changes and track dependencies closely.
```

Comment:
What do you mean by: 

> Modules (see the latest)

I guess we should leave a link to where we want to point the user (?)

Reply:
Yes, I wanted to link to the docs but then I wasn't sure about it since we will need to update the link after each release and then I forgot about it :D